MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,2008 @@
+        h_output_aftermerge_vertex_tkvtxdisterr->Fill(tk_vtx_dist.second.error());
+        h_output_aftermerge_vertex_tkvtxdistsig->Fill(tk_vtx_dist.second.significance());
+      }
+
+      h_output_aftermerge_vertex_chi2->Fill(vchi2);
+      h_output_aftermerge_vertex_ndof->Fill(vndof);
+      h_output_aftermerge_vertex_x->Fill(vx);
+      h_output_aftermerge_vertex_y->Fill(vy);
+      h_output_aftermerge_vertex_rho->Fill(rho);
+      h_output_aftermerge_vertex_phi->Fill(phi);
+      h_output_aftermerge_vertex_z->Fill(vz);
+      h_output_aftermerge_vertex_r->Fill(r);
+
+      for (size_t j = i + 1, je = vertices->size(); j < je; ++j) {
+        const reco::Vertex& vj = vertices->at(j);
+        const double vjx = vj.position().x() - bsx;
+        const double vjy = vj.position().y() - bsy;
+        const double phij = atan2(vjy, vjx);
+        Measurement1D v_dist = vertex_dist(vj, v);
+        h_output_aftermerge_vertex_paird2d->Fill(mag(vx - vjx, vy - vjy));
+        h_output_aftermerge_vertex_paird2dsig->Fill(v_dist.significance());
+        h_output_aftermerge_vertex_pairdphi->Fill(reco::deltaPhi(phi, phij));
+      }
+
+    }
+
+    h_n_at_least_5trk_output_aftermerge_vertices->Fill(count_5trk_vertices);
+
+  }
+
+  //////////////////////////////////////////////////////////////////////
+  // Shared-jet mitigation with the following procedure:
+  //   -   make a set of vertices that have been sorted by ascending number of tracks per vertex
+  //   - 	 loop thru a pair of >=3trk vertices and check whether they share {1,1} and {1,n} shared jets or not 
+  //   -   In the double loop: check one vertex at a time (sv0) and remove a lone track to the jet if it is pointing backward from its vertex (apply dphi < pi/2)
+  //   -   In the double loop: assign a new fitted vertex to the one resolving shared jets
+  //   -   loop thru a set of vertices after the mitigation to clean up a vertex with just one track
+  // Note that:
+  //   - {1,1} shared jets have exactly one track to the jet from both vertices
+  //   - {1,n} shared jets have one of the two vertices contributing exactly one track to the jet
+  //////////////////////////////////////////////////////////////////////
+
+
+  int n_output_aftersharedjets_onetracks = 0;
+
+  if (resolve_shared_jets) {
+    edm::Handle<pat::JetCollection> jets;
+    event.getByToken(shared_jet_token, jets);
+
+    std::vector<std::vector<size_t> > sv_total_track_which_trkidx;
+    std::vector<unsigned int> sv_total_ntrack;
+    std::vector<unsigned int> sv_ascending_total_ntrack;
+    std::vector<size_t> sv_ascending_vtxidx;
+
+    std::vector<std::vector<int> > sv_match_tracktojet_which_trkidx;
+    std::vector<std::vector<int> > sv_match_tracktojet_which_jetidx;
+
+    typedef std::vector<reco::TrackRef> track_vec;
+    const reco::Vertex fake_bs_vtx(beamspot->position(), beamspot->covariance3D());
+
+    std::vector<track_vec> sv_total_track_which_trk_vec;
+
+    std::vector<int> vertex_ntracks;

```

Comment:
Is this ever used anywhere? It doesn't look like it?

Reply:
Yes, you're right. It has no use.

Reply:
Done.